### DISCUSSION TITLE
Paste Behaviour

### DISCUSSION THREAD

**Discussion by Schildkroet** [Orama-Interactive/Pixelorama#721] [Ideas]
Hello,

if i have an image and select and copy a part of it and then insert it in a new image, it gets pasted at the same position of the original image (e.g. select at position 900:900, then it gets pasted at 900:900)

Due to this behaviour, if i copy a part of an big image at a large position and then paste it in a very small image, it gets placed far away from the small image. So i have to scroll out first, search the image, repostion it near the small image and scroll in again.

Q: Is it possible to paste an image at/near cursor position?

**Comment by Variable-ind** [comment]
Not at the moment but it may be possible in the future, (the selection is currently designed this way that it's pasted at the same point it's copied from)

**Comment by OverloadedOrama** [comment]
I changed the paste behavior in a64f102b9a1ba5bf6284f66bd0d3e3dfd5540d22, and the changes will be included in the next version.

Pasted content is now being placed in the middle of the canvas view. Meaning, it will always be pasted in the center of where you are looking. For canvases smaller than the pasted content, it always gets pasted at (0, 0), so you no longer will have to chase the pasted content out of canvas bounds to find it and move it. I added a new option in the Edit menu, "Paste in Place", which preserves the previous behavior. This is pretty much what GIMP does as well.
